### ISSUE TITLE
Not compatible with core-js@3 dependency

### ISSUE THREAD

**Issue by roojay520** [vuejs/vuepress#1529]
- [x] I confirm that this is an issue rather than a question.

#### Version

```json
"vuepress": "^1.0.0-alpha.47",
 "core-js": "3.0.1"
```
#### Steps to reproduce

1. `yarn add vuepress@next core-js`
2. edit `babel.config.js`
```javascript
  presets: [
    [
      '@vue/app',
      {
        useBuiltIns: 'entry',
        corejs: 3,
      },
    ],
  ],
```

#### What is expected?
The project is running.
#### What is actually happening?
Running error:
![image](https://user-images.githubusercontent.com/11495164/55769051-dc1b5200-5ab1-11e9-959d-27d5a16d96ef.png)

If use `core-js@2.6.5` , this docs dev can run, but conflicts with the vue project dependencies `core-js@3.0.1`. How can I fix it?

#### Other relevant information

- Your OS:  Win 10
- Node.js version: v10.15.3
- Browser version: Chrome 73.0.3683.86
- Is this a global or local install?  >local
- Which package manager did you use for the install?  >yarn
- Does this issue occur when all plugins are disabled?

**Comment by hardylinfee** [comment]
same problem

**Comment by adrianjost** [comment]
I guess it's a simple naming issue. core-js renamed `web.dom` to `web.dom-collection` in version 3 [changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)

**Comment by allen-zh** [comment]
+1

when use vuepress with @babel/preset-env@7.4.x (which requires core-js@3), and when core-js under node_modules directly resolves to 3, it will cause vuepress build errors like below.

```
wait Extracting site metadata...
tip Apply theme @vuepress/theme-default ...
tip Apply plugin container (i.e. "vuepress-plugin-container") ...
tip Apply plugin @vuepress/last-updated (i.e. "@vuepress/plugin-last-updated") ...
tip Apply plugin @vuepress/register-components (i.e. "@vuepress/plugin-register-components") ...
tip Apply plugin @vuepress/active-header-links (i.e. "@vuepress/plugin-active-header-links") ...
tip Apply plugin @vuepress/search (i.e. "@vuepress/plugin-search") ...
tip Apply plugin @vuepress/nprogress (i.e. "@vuepress/plugin-nprogress") ...
[11:23:56 AM] Compiling Client
[11:23:56 AM] Compiling Server
[11:23:59 AM] Compiled Server in 3s
[11:24:03 AM] Compiled Client in 7s
(undefined) ./node_modules/@babel/runtime-corejs2/core-js/array/from.js
Module not found: Error: Can't resolve 'core-js/library/fn/array/from' in '/home/travis/build/max-team/Mars/node_modules/@babel/runtime-corejs2/core-js/array'
resolve 'core-js/library/fn/array/from' in '/home/travis/build/max-team/Mars/node_modules/@babel/runtime-corejs2/core-js/array'
  Parsed request is a module
  using description file: /home/travis/build/max-team/Mars/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/array)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /home/travis/build/max-team/Mars/node_modules
        using description file: /home/travis/build/max-team/Mars/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /home/travis/build/max-team/Mars/node_modules/core-js/package.json (relative path: ./library/fn/array/from)
             no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from.jsx doesn't exist
            .vue
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from.vue doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from.json doesn't exist
            .styl
              Field 'browser' doesn't contain a valid alias configuration
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from.styl doesn't exist
            as directory
              /home/travis/build/max-team/Mars/node_modules/core-js/library/fn/array/from doesn't exist
...
```

**Comment by darkskygit** [comment]
temporary fix works fine for me:
npm i @vue/babel-preset-app@next

**Comment by hardylinfee** [comment]
> temporary fix works fine for me:
> npm i @vue/babel-preset-app@next

thanxs

**Comment by jerguslejko** [comment]
same issue here

**Comment by bloqhead** [comment]
@darkskygit this seemed to fix things for me but would you mind sharing your `config.js` and `.babelrc` so I can compare it with mine? I'm trying to understand the issue more thoroughly and it would help me a ton!

**Comment by f3ltron** [comment]
the @darkskygit resolve this issue. Thank's for your question all ! I am closing the issue for now feel free to re open.

**Comment by thedamon** [comment]
This "fix" worked for me once I found it, but for a project that wants to work simply "out of the box", I'm wondering if there's a way we can make it not necessary? 

I installed it in the docs folder of a monorepo project that was not made using the vue-cli (just guessing that's common so the babel-preset most people will be using coming into this situation will satisfy what this repo needs). is there something that ought to be added to peerDependencies though? Because i certainly would not have had any idea what was going on if I hadn't found this issue...
